src/js/select2/core.js

```diff
@@ -77,7 +77,7 @@ define([
     });
 
     // Hide the original select
-    $element.hide();
+    $element.addClass('select2-hidden-accessible');
```

Comment:
It might also be useful to do `$element.attr('aria-hidden', 'true')` here, so screen readers know to not focus on the element.